docs/user-guide/how-to-use-header-only-layoutbuilder.md

```diff
@@ -229,9 +235,56 @@ std::cout << builder.form() << std::endl;
 Passing from C++ to Python
 --------------------------
 
-We want NumPy to own the array buffers, so that they get deleted when the Awkward Array goes out of Python scope, not when the LayoutBuilder goes out of C++ scope. For the hand-off, you can allocate memory for those buffers in Python, presumably with `np.empty(nbytes, dtype=np.uint8)` and get void* pointers to these buffers by casting the output of `numpy_array.ctypes.data` (pointer as integer).
+We want NumPy to own the array buffers, so that they get deleted when the Awkward Array goes out of Python scope, not when the LayoutBuilder goes out of C++ scope. For the hand-off, one can allocate memory for those buffers in Python, presumably with `np.empty(nbytes, dtype=np.uint8)` and get `void*` pointers to these buffers by casting the output of `numpy_array.ctypes.data` (pointer as integer). Then we can pass everything over the border from C++ to Python using e.g. `pybind11`'s `py::buffer_protocol` for the buffers.  
+
+Alternatively, the Python _capsule_ system can be used to tie the lifetime of the allocated buffers to the calling Python scope. `pybind11` makes this fairly trivial, and also permits us to invoke Python code _from_ C++. We can use this approach to call `ak.from_buffers` in order to build an `ak.Array`:
+```cpp
+template <typename T>
+py::object snapshot_builder(const T& builder)
```

Comment:
@ianna I'm assuming we can const-ref the builder, but let me know if I've made an oversight here in terms of best practice!

Reply:
Correct - it's mutable :-)